ddtrace/contrib/anthropic/patch.py

```diff
@@ -0,0 +1,158 @@
+    return None
+
+
+@with_traced_module
+def traced_chat_model_generate(anthropic, pin, func, instance, args, kwargs):
+    chat_messages = get_argument_value(args, kwargs, 0, "messages")
+    integration = anthropic._datadog_integration
+
+    operation_name = func.__name__
+
+    span = integration.trace(
+        pin,
+        "%s.%s.%s" % (instance.__module__, instance.__class__.__name__, operation_name),
+        submit_to_llmobs=True,
+        interface_type="chat_model",
+        provider="anthropic",
+        model=kwargs.get("model", ""),
+        api_key=_extract_api_key(instance),
+    )
+
+    chat_completions = None
+    try:
+        for message_idx, message in enumerate(chat_messages):
+            if isinstance(message, dict):
+                if isinstance(message.get("content", None), str):
+                    if integration.is_pc_sampled_span(span) and message.get("content", "") != "":
+                        span.set_tag_str(
+                            "anthropic.request.messages.%d.content.0.text" % (message_idx),
+                            integration.trunc(message.get("content", "")),
+                        )
+                    span.set_tag_str(
+                        "anthropic.request.messages.%d.content.0.type" % (message_idx),
+                        "text",
+                    )
+                elif isinstance(message.get("content", None), list):
+                    for block_idx, block in enumerate(message.get("content", [])):
+                        if integration.is_pc_sampled_span(span):
+                            if block.get("type", None) == "text" and block.get("text", "") != "":
+                                span.set_tag_str(
+                                    "anthropic.request.messages.%d.content.%d.text" % (message_idx, block_idx),
+                                    integration.trunc(str(block.get("text", ""))),
+                                )
+                            elif block.get("type", None) == "image":
+                                span.set_tag_str(
+                                    "anthropic.request.messages.%d.content.%d.text" % (message_idx, block_idx),
+                                    "([IMAGE DETECTED])",
+                                )
+
+                        span.set_tag_str(
+                            "anthropic.request.messages.%d.content.%d.type" % (message_idx, block_idx),
+                            block.get("type", "text"),
+                        )
+                span.set_tag_str(
+                    "anthropic.request.messages.%d.role" % (message_idx),
+                    message.get("role", ""),
+                )
+        params_to_tag = {k: v for k, v in kwargs.items() if k != "messages"}
+        span.set_tag_str("anthropic.request.parameters", json.dumps(params_to_tag))
+
+        chat_completions = func(*args, **kwargs)
+
+        handle_non_streamed_response(integration, chat_completions, args, kwargs, span)
```

Comment:
Just curious, how does a streamed response get handled in this case? I know we're adding support for that in another PR but wanted to be confident this won't break streamed responses

Reply:
added a test for this